meta-resin-common/recipes-bsp/u-boot/patches/resin-specific-env-configuration.patch

```diff
@@ -48,55 +50,135 @@ index 0000000..2ff0fa5
 +       "resin_uboot_devices=" __stringify(RESIN_UBOOT_DEVICES) "\0" \
 +       "resin_boot_part=" __stringify(RESIN_BOOT_PART) "\0" \
 +       "resin_root_part=" __stringify(RESIN_DEFAULT_ROOT_PART) "\0" \
++       "resin_root_parts=2 3\0" \
++       "resin_bc_file=counter\0" \
++       "resin_counter_size=10\0" \
++       \
++       "resin_load_bc_file=" \
++               "echo Loading ${resin_bc_file} from ${resin_dev_type} device ${resin_dev_index} partition ${resin_root_part_index};" \
++               "${resin_loader} ${resin_dev_type} ${resin_dev_index}:${resin_root_part_index} ${resin_bc_load_value_addr} ${resin_bc_file};\0" \
++       \
++       "resin_set_bc_values=" \
++               "setexpr resin_bc_stored_value_addr ${resin_bc_load_value_addr};" \
++               "setexpr resin_bc_load_value_addr ${resin_bc_load_value_addr} + ${resin_counter_size};" \
++               "setenv resin_bc_root_part ${resin_root_part_index};\0" \
++       \
++       "resin_set_bc_root_part=" \
++               "setexpr resin_bc_load_value_addr ${resin_kernel_load_addr};" \
```

Comment:
There are no math operations here so I would use a simple setenv. As well, this definition seems to be global so I'd move it above along with the other ones in the same "global" context.